ingestion/src/metadata/ingestion/source/dashboard/sigma/client.py

```diff
@@ -0,0 +1,177 @@
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+REST Auth & Client for Sigma
+"""
+
+import traceback
+from base64 import b64encode
+from typing import List, Optional, Tuple
+
+from metadata.generated.schema.entity.services.connections.dashboard.sigmaConnection import (
+    SigmaConnection,
+)
+from metadata.ingestion.ometa.client import REST, ClientConfig
+from metadata.ingestion.source.dashboard.sigma.models import (
+    AuthToken,
+    EdgeSource,
+    EdgeSourceResponse,
+    Elements,
+    ElementsResponse,
+    NodeDetails,
+    OwnerDetails,
+    Workbook,
+    WorkbookDetails,
+    WorkBookPageResponse,
+    WorkBookResponseDetails,
+)
+from metadata.utils.helpers import clean_uri
+from metadata.utils.logger import utils_logger
+
+logger = utils_logger()
+
+
+class SigmaApiClient:
+    """
+    REST Auth & Client for Sigma
+    """
+
+    client: REST
+
+    def __init__(self, config: SigmaConnection):
+        self.config = config
+        token_api_key = str(
+            b64encode(
+                f"{self.config.clientId}:{self.config.clientSecret.get_secret_value()}".encode(
+                    "utf-8"
+                )
+            ).decode("utf-8")
+        )
+
+        headers = {
+            "accept": "application/json",
+            "Content-type": "application/x-www-form-urlencoded",
+        }
+
+        token_config = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            api_version=config.apiVersion,
+            auth_header="Authorization",
+            extra_headers=headers,
+            auth_token=lambda: (token_api_key, 0),
+            auth_token_mode="Basic",
+        )
+
+        self.token_client = REST(token_config)
+
+        client_config = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            api_version=config.apiVersion,
+            auth_token=self.get_auth_token,
+            auth_header="Authorization",
+        )
+
+        self.client = REST(client_config)
+
+    def get_auth_token(self) -> Tuple[str, int]:
+        """
+        generate auth token
+        """
+        payload = {"grant_type": "client_credentials"}
+        result = AuthToken(**self.token_client.post("/auth/token", data=payload))
+        return result.access_token, 0
+
+    def get_dashboards(self) -> Optional[List[Workbook]]:
+        """
+        method to fetch dashboards from api
+        """
+        result = WorkBookResponseDetails(**self.client.get("/workbooks"))
```

Comment:
if client.get returns `None` we'd get an unpacking error here. Maybe we should handle this a bit better + use `model_validate` instead of directly unpacking